exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/trace_exporter/__init__.py

```diff
@@ -0,0 +1,185 @@
+            )
+        )
+        self._compression = compression or _compression_from_env()
+        self._session = requests.Session()
+        self._session.headers.update(self._headers)
+        self._session.headers.update(
+            {"Content-Type": ProtobufEncoder._CONTENT_TYPE}
+        )
+        if self._compression is not Compression.NoCompression:
+            self._session.headers.update(
+                {"Content-Encoding": self._compression.value}
+            )
+        self._shutdown = False
+
+    def _export(self, serialized_data: str):
+        data = serialized_data
+        if self._compression == Compression.Gzip:
+            gzip_data = BytesIO()
+            with gzip.GzipFile(fileobj=gzip_data, mode="w") as gzip_stream:
+                gzip_stream.write(serialized_data)
+            data = gzip_data.getvalue()
+        elif self._compression == Compression.Deflate:
+            data = zlib.compress(bytes(serialized_data))
+
+        return self._session.post(
+            url=self._endpoint,
+            data=data,
+            verify=self._certificate_file,
+            timeout=self._timeout,
+        )
+
+    @staticmethod
+    def _retryable(resp: requests.Response) -> bool:
+        if resp.status_code == 408:
+            return True
+        if resp.status_code >= 500 and resp.status_code <= 599:
+            return True
+        return False
+
+    def export(self, spans) -> SpanExportResult:
+        # After the call to Shutdown subsequent calls to Export are
+        # not allowed and should return a Failure result.
+        if self._shutdown:
+            _logger.warning("Exporter already shutdown, ignoring batch")
+            return SpanExportResult.FAILURE
+
+        serialized_data = ProtobufEncoder.serialize(spans)
+
+        for delay in expo(max_value=self._MAX_RETRY_TIMEOUT):
+
+            if delay == self._MAX_RETRY_TIMEOUT:
+                return SpanExportResult.FAILURE
+
+            resp = self._export(serialized_data)
+            # pylint: disable=no-else-return
+            if resp.status_code in (200, 202):
+                return SpanExportResult.SUCCESS
+            elif self._retryable(resp):
+                _logger.debug(
+                    "Waiting %ss before retrying export of span", delay
+                )
+                sleep(delay)
+                continue
+            else:
+                _logger.warning(
+                    "Failed to export batch code: %s, reason: %s",
+                    resp.status_code,
+                    resp.text,
+                )
+                return SpanExportResult.FAILURE
+        return SpanExportResult.FAILURE
+
+    def shutdown(self):
+        if self._shutdown:
+            _logger.warning("Exporter already shutdown, ignoring call")
+            return
+        self._session.close()
+        self._shutdown = True
+
+
+def _headers_from_env() -> Optional[Dict[str, str]]:
```

Comment:
No need to necessarily change this here, but i think we're repeating the pattern of splitting variables by `,` and `=` in a few places, might be nice to pull that code into a utility.

Reply:
Agree. Probably some internal only private method in api/sdk.

Reply:
+1 also what happens if the user wants to put a `,` or `=` character in their header/regex whatever. Is an escape sequence documented anywhere?

Reply:
I believe it is documented in the specification.

Reply:
Created an issue here https://github.com/open-telemetry/opentelemetry-python/issues/1949.